src/main/java/leo/client/Browser.java

```diff
@@ -20,30 +20,31 @@ public class Browser {
 
     public static void open(String url) {
         String osName = System.getProperty("os.name");
-
+        Runtime runtime = Runtime.getRuntime();
         try {
             if (osName.startsWith("Mac OS")) {
-                Class fileMgr = Class.forName("com.apple.eio.FileManager");
-                Method openURL = fileMgr.getDeclaredMethod("openURL", String.class);
-                openURL.invoke(null, url);
-            } else if (osName.startsWith("Windows"))
-                Runtime.getRuntime().exec("rundll32 url.dll,FileProtocolHandler " + url);
+            	runtime.exec("open " + url);
+            } else if (osName.startsWith("Windows")) {
+            	runtime.exec("rundll32 url.dll,FileProtocolHandler " + url);
+            }
             else {
                 //assume Unix or Linux
                 String[] browsers =
                         {"firefox", "opera", "konqueror", "epiphany", "mozilla", "netscape"};
                 String browser = null;
                 for (int count = 0; count < browsers.length && browser == null; count++)
-                    if (Runtime.getRuntime().exec(new String[]{"which", browsers[count]}).waitFor() == 0)
+                    if (Runtime.getRuntime().exec(new String[]{"which", browsers[count]}).waitFor() == 0) {
                         browser = browsers[count];
-
-                if (browser == null)
-                    throw new Exception("Could not find web browser");
-                else
-                    Runtime.getRuntime().exec(new String[]{browser, url});
+                    }
+                if (browser == null) {
+                    throw new Exception(String.format("Could not find web browser.%nURL: %s", url));
+                }
+                else {
+                	runtime.exec("xdg-open " + url);
+                }
             }
         } catch (Exception e) {
             JOptionPane.showMessageDialog(null, errMsg + ":\n" + e.getLocalizedMessage());
         }
     }
-} 
+} 
```

Comment:
Removing newline EOF character?

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

Reply:
Not really sure how to fix this.
In my Eclipse IDE it shows as 50 lines and no new line.

I tried loading it VSC and it shows a smaller amount of line but again no EOF.